pkg/kubelet/kubelet_test.go

```diff
@@ -2154,6 +2156,65 @@ func TestMakeEnvironmentVariables(t *testing.T) {
 				"KUBERNETES_RO_PORT_8087_TCP_ADDR=1.2.3.7"),
 			21,
 		},
+		{
+			"simple substitution",
+			&api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "unknown", Name: "test-pod"}},
+			&api.Container{
+				Env: []api.EnvVar{
+					{Name: "BAR", Value: "bar"},
+					{Name: "FOO", Value: "${BAR}"},
+				},
+			},
+			"kubernetes",
+			true,
+			util.NewStringSet(
+				"BAR=bar",
+				"FOO=bar",
+			),
+			2,
+		},
+		{
+			"slightly more complex substitution",
+			&api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "unknown", Name: "test-pod"}},
+			&api.Container{
+				Env: []api.EnvVar{
+					{Name: "BAR", Value: "bar"},
```

Comment:
@bgrant0607 Disregard, I see what you mean.  That is a good point re: consuming the API within the image, and another facet of how hard it would be make a sensible API to consume that way.